app/scripts/controllers/projectBrowseCatalog.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('ProjectBrowseCatalogController',
+    function($scope,
+             $location,
+             $routeParams,
+             AuthorizationService,
+             AuthService,
+             CatalogService) {
+      AuthService.withUser().then(function() {
+        var currentProjectName = $routeParams.project;
+        if (!currentProjectName) {
+          $location.url('/');
+          return;
+        }
+
+        AuthorizationService.getProjectRules(currentProjectName).then(function () {
+          var canIAddToProject = AuthorizationService.canIAddToProject(currentProjectName);
+          if (!canIAddToProject) {
+            $location.url('project/' + currentProjectName);
+            return;
+          }
+
+          CatalogService.getCatalogItems().then(function (items) {
+            $scope.catalogItems = items;
```

Comment:
It would be nice to show the image streams and templates from the current project as well since we have a project context.

Reply:
We don't currently support browsing and choosing image streams. I can add project templates though.

Reply:
> We don't currently support browsing and choosing image streams.

We support builder images in the catalog itself.